src/librustc/traits/specialize/specialization_graph.rs

```diff
@@ -113,6 +113,11 @@ impl<'a, 'gcx, 'tcx> Children {
                                                         possible_sibling,
                                                         impl_def_id);
                 if let Some(impl_header) = overlap {
+                    if tcx.impl_always_allowed_to_overlap(impl_def_id)
+                        && tcx.impl_always_allowed_to_overlap(possible_sibling) {
+                        return Ok((true, true));
```

Comment:
I guess it doesn't matter, but I think `Ok((false, false))` would be a better return. It'll do the same thing either way though.